src/Writer/PhpArray.php

```diff
@@ -139,4 +158,54 @@ protected function processIndented(array $config, array $arraySyntax, &$indentLe
 
         return $arrayString;
     }
+
+    /**
+     * Process a string configuration value
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function processStringValue($value)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($value)) {
+            return $value . '::class';
+        }
+
+        return var_export($value, true);
+    }
+
+    /**
+     * Process a string configuration key
+     *
+     * @param string $key
+     * @return string
+     */
+    protected function processStringKey($key)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($key)) {
+            return $key . '::class';
+        }
+
+        return "'" . addslashes($key) . "'";
+    }
+
+    /**
+     * Check whether a string represents a resolvable FQCN
+     *
+     * @param string $string
+     * @return bool
+     */
+    protected function checkStringIsFqn($string)
```

Comment:
> This regex only works with single name components in different contexts.

You can adapt the regex to match multiple backslashes:

`'/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)*$/'`

See http://regexper.com/#%2F^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*%28%3F%3A\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*%29*%24%2F